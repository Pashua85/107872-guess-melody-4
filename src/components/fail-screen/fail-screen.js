import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {restartGame} from '../../action-creators/action-creators';
import {Redirect} from 'react-router-dom';

const FailScreen = (props) => {
  const {onAgainClick, step} = props;

  if (step === 0) {
    return <Redirect to="/dev-artist" />;
  }

  return (
    <section className="result">
      <div className="result__logo">
        <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" />
      </div>
      <h2 className="result__title">Какая жалость!</h2>
      <p className="result__total result__total--fail">У вас закончились все попытки. Ничего, повезёт в следующий раз!</p>
      <button
        className="replay"
        type="button"
        onClick={onAgainClick}
      >
        Попробовать ещё раз
      </button>
    </section>
  );
};

FailScreen.propTypes = {
  onAgainClick: PropTypes.func.isRequired,
  step: PropTypes.number.isRequired
};

const mapStateToProps = (state) => ({
  step: state.step
});

const mapDispatchToProps = (dispatch) => ({
  onAgainClick: () => {
    dispatch(restartGame());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(FailScreen);
export {FailScreen};


