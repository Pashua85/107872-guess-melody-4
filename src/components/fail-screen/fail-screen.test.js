import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {FailScreen} from './fail-screen';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`FailScreen`, () => {
  it(`should render correctly`, () => {
    const wrapper = shallow(
        <FailScreen
          step={1}
          onAgainClick={() => {}}
        />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
