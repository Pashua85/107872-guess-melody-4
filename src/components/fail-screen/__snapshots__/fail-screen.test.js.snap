// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FailScreen should render correctly 1`] = `ShallowWrapper {}`;
